### DISCUSSION TITLE
no src after bin/download

### DISCUSSION THREAD

**Discussion by guidobuldrighini** [markshust/docker-magento#552] [Q&A]
Hello,
I tried to install a new magento project (2.4.3-p1), either by trying with the one-line installation or with the manual installation of the new project I get the following log and the src folder is not created.
If, on the other hand, I use the guide to import an existing project, everything works fine.
I have tried with both docker on linux and windows with docker on wsl.
Below is the final part of the log:

487/487 [============================] 100%
    ...Module Magento_InventoryApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryApi module
    ...Module Magento_InventorySalesApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventorySalesApi module
    ...Module Magento_InventoryCatalogApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryCatalogApi module
    ...Module Magento_InventoryAdvancedCheckout recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryAdvancedCheckout module
    ...Module Magento_InventoryBundleImportExport recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryBundleImportExport module
    ...Module Magento_InventoryConfigurationApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryConfigurationApi module
    ...Module Magento_InventoryBundleProduct recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryBundleProduct module
    ...Module Magento_InventorySourceSelectionApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventorySourceSelectionApi module
    ...Module Magento_Inventory recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_Inventory module
    ...Module Magento_InventorySourceDeductionApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventorySourceDeductionApi module
    ...Module Magento_InventoryReservationsApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryReservationsApi module
    ...Module Magento_InventorySales recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventorySales module
    ...Module Magento_InventoryMultiDimensionalIndexerApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryMultiDimensionalIndexerApi module
    ...Module Magento_InventoryIndexer recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryIndexer module
    ...Module Magento_InventoryCatalogAdminUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryCatalogAdminUi module
    ...Module Magento_InventoryBundleProductAdminUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryBundleProductAdminUi module
    ...Module Magento_InventoryBundleProductIndexer recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryBundleProductIndexer module
    ...Module Magento_InventoryCache recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryCache module
    ...Module Magento_InventoryCatalogSearch recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryCatalogSearch module
    ...Module Magento_InventoryCatalogSearchBundleProduct recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryCatalogSearchBundleProduct module
    ...Module Magento_InventoryCatalogSearchConfigurableProduct recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryCatalogSearchConfigurableProduct module
    ...Module Magento_InventoryConfigurableProduct recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryConfigurableProduct module
    ...Module Magento_InventoryConfigurableProductAdminUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryConfigurableProductAdminUi module
    ...Module Magento_InventoryConfigurableProductFrontendUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryConfigurableProductFrontendUi module
    ...Module Magento_InventoryConfigurableProductIndexer recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryConfigurableProductIndexer module
    ...Module Magento_InventoryDistanceBasedSourceSelectionApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryDistanceBasedSourceSelectionApi module
    ...Module Magento_InventoryDistanceBasedSourceSelection recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryDistanceBasedSourceSelection module
    ...Module Magento_InventoryDistanceBasedSourceSelectionAdminUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryDistanceBasedSourceSelectionAdminUi module
    ...Module Magento_InventoryElasticsearch recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryElasticsearch module
    ...Module Magento_InventoryExportStockApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryExportStockApi module
    ...Module Magento_InventoryConfiguration recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryConfiguration module
    ...Module Magento_InventoryExportStock recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryExportStock module
    ...Module Magento_InventoryCatalog recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryCatalog module
    ...Module Magento_InventoryGraphQl recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryGraphQl module
    ...Module Magento_InventoryGroupedProduct recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryGroupedProduct module
    ...Module Magento_InventoryGroupedProductAdminUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryGroupedProductAdminUi module
    ...Module Magento_InventoryGroupedProductIndexer recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryGroupedProductIndexer module
    ...Module Magento_InventoryImportExport recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryImportExport module
    ...Module Magento_InventoryInStorePickupApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupApi module
    ...Module Magento_InventoryInStorePickup recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickup module
    ...Module Magento_InventoryAdminUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryAdminUi module
    ...Module Magento_InventoryInStorePickupAdminUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupAdminUi module
    ...Module Magento_InventoryInStorePickupShippingApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupShippingApi module
    ...Module Magento_InventoryInStorePickupSalesApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupSalesApi module
    ...Module Magento_InventoryInStorePickupFrontend recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupFrontend module
    ...Module Magento_InventoryInStorePickupGraphQl recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupGraphQl module
    ...Module Magento_InventoryInStorePickupMultishipping recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupMultishipping module
    ...Module Magento_InventoryInStorePickupQuoteGraphQl recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupQuoteGraphQl module
    ...Module Magento_InventoryInStorePickupSales recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupSales module
    ...Module Magento_InventoryInStorePickupQuote recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupQuote module
    ...Module Magento_InventoryInStorePickupSalesAdminUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupSalesAdminUi module
    ...Module Magento_InventoryInStorePickupShipping recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupShipping module
    ...Module Magento_InventoryInStorePickupShippingAdminUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupShippingAdminUi module
    ...Module Magento_InventoryInStorePickupWebapiExtension recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryInStorePickupWebapiExtension module
    ...Module Magento_InventoryLowQuantityNotificationApi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryLowQuantityNotificationApi module
    ...Module Magento_InventoryLowQuantityNotification recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryLowQuantityNotification module
    ...Module Magento_InventoryLowQuantityNotificationAdminUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryLowQuantityNotificationAdminUi module
    ...Module Magento_InventoryProductAlert recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryProductAlert module
    ...Module Magento_InventoryQuoteGraphQl recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryQuoteGraphQl module
    ...Module Magento_InventoryRequisitionList recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryRequisitionList module
    ...Module Magento_InventoryReservationCli recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryReservationCli module
    ...Module Magento_InventoryReservations recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryReservations module
    ...Module Magento_InventorySalesAdminUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventorySalesAdminUi module
    ...Module Magento_InventoryCatalogFrontendUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryCatalogFrontendUi module
    ...Module Magento_InventorySalesFrontendUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventorySalesFrontendUi module
    ...Module Magento_InventorySetupFixtureGenerator recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventorySetupFixtureGenerator module
    ...Module Magento_InventoryShipping recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryShipping module
    ...Module Magento_InventoryShippingAdminUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryShippingAdminUi module
    ...Module Magento_InventorySourceSelection recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventorySourceSelection module
    ...Module Magento_InventorySwatchesFrontendUi recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventorySwatchesFrontendUi module
    ...Module Magento_InventoryVisualMerchandiser recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryVisualMerchandiser module
    ...Module Magento_InventoryWishlist recognized as part of Magento Multi Source Inventory implementation
    ...No special rule applied for Magento_InventoryWishlist module
121 package suggestions were added by new dependencies, use `composer suggest` to see details.
Package container-interop/container-interop is abandoned, you should avoid using it. Use psr/container instead.
Package csharpru/vault-php-guzzle6-transport is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/consistency is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/console is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/event is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/exception is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/file is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/iterator is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/protocol is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/stream is abandoned, you should avoid using it. No replacement was suggested.
Package hoa/ustring is abandoned, you should avoid using it. No replacement was suggested.
Package phpunit/php-token-stream is abandoned, you should avoid using it. No replacement was suggested.
Package sebastian/resource-operations is abandoned, you should avoid using it. No replacement was suggested.
Generating autoload files
120 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
PHP CodeSniffer Config installed_paths set to ../../magento/magento-coding-standard,../../phpcompatibility/php-compatibility

**Comment by markshust** [comment]
I just experienced this myself! There was a bug in the onelinesetup script. It should work now!

**Reply by okidevlux** [reply]
Hi, got the same issue with 2.4.3, also got a flag about 8.1.6 php instead of 7.4, i would like to run it with 7.4 for a module compatibility, any idea how to change the php in docker ?
i tried to edit the compose yam file , or any other file to edit ?

**Reply by JesperHerrloff** [reply]
Hope you have solved it by now. You could select some older php for example markoshust/magento-php:7.4-fpm-9

You will se a whole list of tags/version in this link https://hub.docker.com/r/markoshust/magento-php/tags

**Comment by YevhenZvieriev** [comment]
### It helps me every time:

1. **docker exec -it magento-phpfpm-1 bash**
For your first command, replace magento-phpfpm-1 with your actual docker name you can get by running docker container ls. In my case, it was magento-phpfpm-1, so I ran docker exec -it magento-phpfpm-1 bash.
This logs you into the bash shell of the docker image.

2. **composer create-project --repository-url=https://repo.magento.com/ magento/project-community-edition=2.4.3 .**
If needed, change 2.4.3 to the version that you want.

3. ### mv project-community-edition/* .
   Note: Not sure if needed

4. **exit**
 
5. **bin/magento**

**Reply by YPOEL** [reply]
@YevhenZvieriev i have been busy for the last week getting the magento docker up and running on my Macbook air with M1 CPU.
I was trying your suggestion, i could only connect to the docker container shell via its container ID, but after that i got the error:

app@9f8a48d94813:~/html$ composer create-project --repository-url=https://repo.magento.com/ magento/project-community-edition=2.4.3
Creating a "magento/project-community-edition=2.4.3" project at "./project-community-edition"
Warning from repo.magento.com: Your Magento authentication keys are invalid. Please double-check your keys in your Marketplace account. For instructions, visit https://devdocs.magento.com/guides/v2.3/install-gde/prereq/connect-auth.html
Warning from repo.magento.com: Your Magento authentication keys are invalid. Please double-check your keys in your Marketplace account. For instructions, visit https://devdocs.magento.com/guides/v2.3/install-gde/prereq/connect-auth.html

In AuthHelper.php line 216:
                                                                               
  Invalid credentials (HTTP 401) for 'https://repo.magento.com/packages.json'  
  , aborting.                                                                  
                                                                               

create-project [-s|--stability STABILITY] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--repository REPOSITORY] [--repository-url REPOSITORY-URL] [--add-repository] [--dev] [--no-dev] [--no-custom-installers] [--no-scripts] [--no-progress] [--no-secure-http] [--keep-vcs] [--remove-vcs] [--no-install] [--no-audit] [--audit-format AUDIT-FORMAT] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--ask] [--] [<package> [<directory> [<version>]]]

**Reply by YevhenZvieriev** [reply]
I see you have a problem with Magento authentication keys.

Check [this](https://github.com/markshust/docker-magento/issues/1028#issuecomment-1868334008) to resolve this issue.

**Reply by YPOEL** [reply]
@YevhenZvieriev this is great, adding the repo keys to the nano ~/.composer/auth.json file got me on the right path.

I forgot to do this, even the single setup script now works right out of the box!
Thanks a million!!

**Reply by YevhenZvieriev** [reply]
I'm happy to hear that you got it working. 

BTW, you don't need to add the repo keys to `auth.json` file manually,  usually, you can do this during the first execution of the one-line script.

Let me know if you need anything else.

**Reply by YPOEL** [reply]
Ok good to konw, probably missed this when running the script for the first time!

**Reply by Real-Rabbitdigital** [reply]
Hi all, I get this at the end:

Composer could not find a composer.json file in /Users/.../Sites/magento
To initialize a project, please create a composer.json file. See https://getcomposer.org/basic-usage

I checked the composer version:

Warrens-iMac:magento ...$ composer -v
   ______
  / ____/___  ____ ___  ____  ____  ________  _____
 / /   / __ \/ __ `__ \/ __ \/ __ \/ ___/ _ \/ ___/
/ /___/ /_/ / / / / / / /_/ / /_/ (__  )  __/ /
\____/\____/_/ /_/ /_/ .___/\____/____/\___/_/
                    /_/
Composer version 2.7.2 2024-03-11 17:12:18

I didn't get anything on SSL setup so I'm getting nothing in https://magento.test/ right now.

I checked the Magento version and I get this:

Warrens-iMac:magento ...$ bin/magento -V
Magento CLI 2.4.6-p4

Anybody got any ideas?
Thanks, Warren

**Reply by YevhenZvieriev** [reply]
Are you sure, that `composer.json` exists within the Docker container?
 
Try to execute the following commands:

```
bin/copytocontainer --all
bin/composer install
```

You should use `bin/composer` instead of `composer`.

**Reply by Real-Rabbitdigital** [reply]
Thanks @YevhenZvieriev, yes I followed the manual installation and that worked. 

It was an issue with the SSL not installing first time round.
